server/application/application.go

```diff
@@ -531,6 +531,45 @@ func (s *Server) GetManifests(ctx context.Context, q *application.ApplicationMan
 			if q.GetRevision() != "" {
 				source.TargetRevision = q.GetRevision()
 			}
+			// When using sourceHydrator with different repos, check if the requested revision
+			// matches DrySHA or HydratedSHA and use the appropriate source config from Status.
+			// Uses Status (not Spec) because commits were created with that config at hydration time.
```

Comment:
I think we'd discussed removing the inference logic and just always using either dry or hydrated based on a param (with default = dry for an empty param)? Maybe an old commit got pushed?

Reply:
yep, i'm sorry I somehow messed up and pushed an old branch.. sorting it out right now!